db/cryptodata/plugin.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cryptodata
+
+import (
+	"io/ioutil"
+	"encoding/pem"
+	"crypto/x509"
+	"errors"
+	"github.com/ligato/cn-infra/infra"
+)
+
+// Config is used to read private key from file
+type Config struct {
+	// Private key file is used to create rsa.PrivateKey from this PEM path
+	PrivateKeyFiles []string `json:"private-key-files"`
+}
+
+// Deps lists dependencies of the cryptodata plugin.
+type Deps struct {
+	infra.Deps
+}
+
+// Plugin implements cryptodata as plugin.
+type Plugin struct {
+	Deps
+	// Client provides crypto support
+	*Client
+	// Plugin is disabled if there is no config file available
+	disabled bool
+}
+
+// Init initializes cryptodata plugin.
+func (plugin *Plugin) Init() (err error) {
+	var config Config
+	found, err := plugin.PluginConfig.GetValue(&config)
+	if err != nil {
+		return err
+	}
+
+	if !found {
+		plugin.Log.Info("cryptodata config not found, skip loading this plugin")
+		plugin.disabled = true
+		return nil
+	}
+
+	// Read client config and create it
+	clientConfig := ClientConfig{}
+	for _, file := range config.PrivateKeyFiles {
+		bytes, err := ioutil.ReadFile(file)
+		if err != nil {
+			return err
+		}
+
+		block, rest := pem.Decode(bytes)
+		if block == nil {
+			return errors.New("failed to decode PEM for key " + file)
+		}
+
+		for {
```

Comment:
This for loop should be defined as `for block != nil {` otherwise the call on line 73 can crash by accessing nil pointer if the previous iteration failed to decode anything from rest.